signac/contrib/project.py

```diff
@@ -681,6 +681,15 @@ def groupby(self, key=None, default=None):
             A default value to be used when a given state point key is not present (must
             be sortable).
         """
+        if default is None:
+            if isinstance(key, str):
+                _filter = {key: {"$exists": True}}
+            elif isinstance(key, Iterable):
+                _filter = {k: {"$exists": True} for k in key}
+            else:
+                _filter = None
+            return self.find_jobs(filter=_filter).groupby(key, default=default)
```

Comment:
I'm wondering, would it make more sense to implement this logic directly within the `groupby()` implementation of the `JobsCursor` class? So that it would also work when users call this directly on a result of `find_jobs()` ?

Reply:
Yes, that's a good idea. We try to keep as much logic in `JobsCursor` as possible, so that it's useful for operations on subsets of the project.